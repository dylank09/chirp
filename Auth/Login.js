import React, { useState } from "react";
import {
  StyleSheet,
  Image,
  Text,
  View,
  SafeAreaView,
  KeyboardAvoidingView,
} from "react-native";

import { theme } from "../assets/Theme";
import logo from "../assets/logo.png";
import ChirpButton from "../components/ChirpButton";
import AuthTextBox from "../components/AuthTextBox";
import AuthAlert from "../components/AuthAlert";
import AuthProviderButton from "../components/AuthProviderButton";

import firebase from "firebase/app";
import "firebase/auth";
import GoogleSignIn from "./GoogleSignIn";
import MicrosoftSignIn from "./MicrosoftSignIn";

export default function Login({ navigation }) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [emailError, setEmailError] = useState("");
  const [passwordError, setPasswordError] = useState("");

  const next = "ChirpGroups";

  function signInEmailPass() {
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then((userCredential) => {
        // Signed in
        // var user = userCredential.user;
        navigation.navigate("ChirpGroups");
        // set currentUser in the App to some value in order to render the main screen?
      })
      .catch((error) => {
        console.log(error);
        if (error.code == "auth/invalid-email")
          setEmailError("Email address is not valid.");
        else if (error.code == "auth/user-not-found")
          setEmailError("User not found.");
        else if (error.code == "auth/wrong-password")
          setPasswordError("Incorrect credentials given.");
        else {
          setEmailError("Incorrect details. Try again.");
        }
      });
  }

  function navToRegister() {
    navigation.navigate("Register");
  }

  return (
    <SafeAreaView style={styles.container}>
      <Image style={styles.logo} source={logo}></Image>
      <Text style={styles.headingText}>Sign In</Text>

      <KeyboardAvoidingView style={styles.inputContainer} behavior="padding">
        <AuthTextBox placeholder="Email" onChangeText={setEmail}></AuthTextBox>
        <AuthAlert text={emailError}></AuthAlert>
        <AuthTextBox
          placeholder="Password"
          onChangeText={setPassword}
        ></AuthTextBox>
        <AuthAlert text={passwordError}></AuthAlert>
        <ChirpButton text="Continue" onPress={signInEmailPass}></ChirpButton>
      </KeyboardAvoidingView>
      <Text style={styles.orText}>or</Text>
      <View style={styles.line} />
      <AuthProviderButton
        text="Sign in with Google"
        type="google"
        onPress={() => GoogleSignIn(navigation, next)}
      ></AuthProviderButton>
      <AuthProviderButton
        text="Sign in with Microsoft"
        type="microsoft"
        onPress={() => MicrosoftSignIn(navigation, next)}
      ></AuthProviderButton>
      <Text style={styles.registerText} onPress={navToRegister}>
        Register new account
      </Text>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    backgroundColor: theme.colors.background,
    alignItems: "center",
    justifyContent: "center",
  },
  inputContainer: {
    flex: 2,
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
    marginBottom: 20,
  },
  headingText: {
    flex: 1,
    fontSize: 28,
    color: theme.colors.text,
    marginBottom: 25,
    marginTop: 10,
  },
  logo: {
    width: 130,
    height: 130,
    padding: 0,
    marginTop: 10,
    marginBottom: 15,
  },
  orText: {
    fontSize: theme.dimensions.standardFontSize,
    color: theme.colors.text,
  },
  line: {
    borderBottomColor: theme.colors.text,
    borderBottomWidth: 1,
    marginBottom: 15,
    width: "50%",
  },
  registerText: {
    flex: 0.3,
    color: theme.colors.placeholderColor,
    fontSize: theme.dimensions.standardFontSize,
    marginTop: 10,
    marginBottom: 10,
    textDecorationLine: "underline",
  },
});
