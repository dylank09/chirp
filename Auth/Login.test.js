import React from "react";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";

import Login from "./Login";

describe("<Login />", () => {
  const tree = renderer.create(<Login />).toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("keyboard avoid view contains five children", () => {
    expect(tree.children[2].children).toHaveLength(5);
  });

  it("continue login button is present", () => {
    const { getByTestId } = render(<Login />);
    const foundButton = getByTestId("loginButton");
    expect(foundButton).toBeTruthy();
  });

  it("Google button is present", () => {
    const { getByTestId } = render(<Login />);
    const foundButton = getByTestId("googleButton");
    expect(foundButton).toBeTruthy();
  });

  it("navigate to register button is present", () => {
    const { getByTestId } = render(<Login />);
    const foundButton = getByTestId("navToRegisterButton");
    expect(foundButton).toBeTruthy();
  });
});
