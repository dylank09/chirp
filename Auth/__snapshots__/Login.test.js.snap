// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(76, 76, 71)",
      "flex": 1,
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 120,
        "marginBottom": 15,
        "padding": 0,
        "width": 120,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
        "fontSize": 28,
        "marginBottom": 15,
      }
    }
  >
    Sign In
  </Text>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 40,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={true}
      multiline={false}
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="rgb(158, 158, 158)"
      rejectResponderTermination={true}
      secureTextEntry={false}
      selectTextOnFocus={false}
      selectionColor="rgb(193, 73, 83)"
      style={
        Array [
          Object {
            "borderBottomColor": "rgb(255, 255, 255)",
            "borderBottomWidth": 1,
            "color": "rgb(255, 255, 255)",
            "width": "82%",
          },
          Object {
            "fontSize": 15,
            "marginBottom": 4,
            "marginTop": 7,
            "padding": 5,
          },
        ]
      }
      testID="chirpTextBox"
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "rgb(193, 73, 83)",
          "fontSize": 11,
          "height": 15,
          "margin": 0,
          "paddingLeft": 3,
          "width": "75%",
        }
      }
    >
       
    </Text>
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={true}
      multiline={false}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="rgb(158, 158, 158)"
      rejectResponderTermination={true}
      secureTextEntry={true}
      selectTextOnFocus={true}
      selectionColor="rgb(193, 73, 83)"
      style={
        Array [
          Object {
            "borderBottomColor": "rgb(255, 255, 255)",
            "borderBottomWidth": 1,
            "color": "rgb(255, 255, 255)",
            "width": "82%",
          },
          Object {
            "fontSize": 15,
            "marginBottom": 4,
            "marginTop": 7,
            "padding": 5,
          },
        ]
      }
      testID="chirpTextBox"
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "rgb(193, 73, 83)",
          "fontSize": 11,
          "height": 15,
          "margin": 0,
          "paddingLeft": 3,
          "width": "75%",
        }
      }
    >
       
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(193, 73, 83)",
          "borderRadius": 18,
          "height": undefined,
          "margin": 10,
          "opacity": 1,
          "padding": 10,
          "width": "40%",
        }
      }
      testID="loginButton"
    >
      <Text
        style={
          Object {
            "color": "rgb(255, 255, 255)",
            "fontSize": 15,
          }
        }
      >
        Continue
         
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "borderBottomColor": "rgb(255, 255, 255)",
        "borderBottomWidth": 1,
        "color": "rgb(255, 255, 255)",
        "fontSize": 15,
        "textAlign": "center",
        "width": "40%",
      }
    }
  >
    or
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 18,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 8,
        "opacity": 1,
        "paddingBottom": 7,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 7,
        "width": "70%",
      }
    }
    testID="googleButton"
  >
    <Image
      source={1}
      style={
        Object {
          "height": 19,
          "margin": 3,
          "width": 19,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontWeight": "600",
          "marginLeft": 5,
        }
      }
    >
      Sign in with Google
       
    </Text>
  </View>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "rgb(158, 158, 158)",
        "fontSize": 15,
        "marginBottom": 10,
        "textDecorationLine": "underline",
      }
    }
    testID="navToRegisterButton"
  >
    Register new account
  </Text>
</RCTSafeAreaView>
`;
